packages/x-date-pickers-pro/src/internals/utils/date-range-manager.ts

```diff
@@ -31,7 +31,8 @@ export function calculateRangeChange({
   shouldMergeDateAndTime = false,
   referenceDate,
 }: CalculateRangeChangeOptions): CalculateRangeChangeResponse {
-  const [start, end] = range;
+  const start = range[0] == null || !utils.isValid(range[0]) ? null : range[0];
+  const end = range[1] == null || !utils.isValid(range[1]) ? null : range[1];
```

Comment:
couldn't we reverse the condition here? I did not check, but I think `isValid` on the adapters already include a `if (value == null) return false`, no?

```suggestion
  const start = utils.isValid(range[0]) ? range[0] : null;
  const end = utils.isValid(range[1]) ? range[1] : null;
```

Reply:
With a small change on the typing of `isValid` we can yes

Fixed